bundles/org.openhab.binding.livisismarthome/src/main/java/org/openhab/binding/livisismarthome/internal/handler/LivisiDeviceHandler.java

```diff
@@ -0,0 +1,1021 @@
+                        autoModeCommand, CapabilityStateDTO.STATE_VALUE_OPERATION_MODE_AUTO,
+                        CapabilityStateDTO.STATE_VALUE_OPERATION_MODE_MANUAL);
+            }
+        }
+    }
+
+    private void commandSwitchAlarm(Command command, LivisiBridgeHandler bridgeHandler) {
+        if (command instanceof OnOffType) {
+            bridgeHandler.commandSwitchAlarm(deviceId, OnOffType.ON.equals(command));
+        }
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing LIVISI SmartHome device handler.");
+        initializeThing(isBridgeOnline());
+    }
+
+    @Override
+    public void dispose() {
+        unregisterListeners(bridgeHandler, deviceId);
+    }
+
+    private static void unregisterListeners(@Nullable LivisiBridgeHandler bridgeHandler, String deviceId) {
+        if (bridgeHandler != null) {
+            bridgeHandler.unregisterDeviceStatusListener(deviceId);
+        }
+    }
+
+    @Override
+    public void bridgeStatusChanged(final ThingStatusInfo bridgeStatusInfo) {
+        logger.debug("bridgeStatusChanged {}", bridgeStatusInfo);
+        initializeThing(ThingStatus.ONLINE == bridgeStatusInfo.getStatus());
+    }
+
+    /**
+     * Initializes the {@link Thing} corresponding to the given status of the bridge.
+     * 
+     * @param isBridgeOnline true if the bridge thing is online, otherwise false
+     */
+    private void initializeThing(final boolean isBridgeOnline) {
+        logger.debug("initializeThing thing {} bridge online status: {}", getThing().getUID(), isBridgeOnline);
+        final String configDeviceId = (String) getConfig().get(PROPERTY_ID);
+        if (configDeviceId != null) {
+            deviceId = configDeviceId;
+
+            Optional<LivisiBridgeHandler> bridgeHandler = registerAtBridgeHandler();
+            if (bridgeHandler.isPresent()) {
+                if (isBridgeOnline) {
+                    initializeProperties();
+
+                    Optional<DeviceDTO> deviceOptional = getDevice();
+                    if (deviceOptional.isPresent()) {
+                        DeviceDTO device = deviceOptional.get();
+                        if (device.isReachable() != null && !device.isReachable()) {
+                            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                    "@text/error.notReachable");
+                        } else {
+                            updateStatus(ThingStatus.ONLINE);
+                        }
+                    } else {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.GONE, "@text/error.deviceNotFound");
+                    }
+                } else {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+                }
+            } else {
+                updateStatus(ThingStatus.OFFLINE);
```

Comment:
If this is the case of missing bridge, this should be `ThingStatusDetail.CONFIGURATION_ERROR`.

Reply:
ok, solved